portal/src/views/shared/Comments.vue

```diff
@@ -0,0 +1,354 @@
+            <button type="submit" class="new-comment-submit" v-if="newComment.body"> Post </button>
+        </form>
+
+        <div v-if="errorGetComments"> Something went wrong loading the comments. </div>
+        <div v-if="errorPostComment"> Something went saving your comment. </div>
+
+        <div class="list" v-if="posts.length > 0">
+            <div class="subheader">
+                <span class="comments-counter" v-if="viewType === 'project'"> {{posts.length}} comments </span>
+                <header v-if="viewType === 'data'"> Notes & Comments </header>
+            </div>
+            <transition-group name="fade">
+                <div class="comment comment-main" v-for="post in posts" v-bind:key="post.id">
+                    <UserPhoto :user="post.author"> </UserPhoto>
+                    <div class="flex column">
+                        <span class="timestamp"> {{formatTimestamp(post.createdAt)}} </span>
+                        <span class="author"> {{post.author.name}} </span>
+                        <span class="body"> {{post.body}} </span>
+
+                        <transition-group name="fade">
+                            <div class="comment" v-for="reply in post.replies" v-bind:key="reply.id">
+                                <UserPhoto :user="post.author"> </UserPhoto>
+                                <div class="flex column">
+                                    <span class="author"> {{reply.author.name}} </span>
+                                    <span class="body"> {{reply.body}} </span>
+                                </div>
+                            </div>
+                        </transition-group>
+
+                        <transition name="fade">
+                            <form @submit.prevent="save(newReply)" class="new-comment reply" v-if="newReply && newReply.threadId === post.id">
+                                <UserPhoto :user="user"> </UserPhoto>
+                                <input type="text" placeholder="Reply to comment" v-model="newReply.body">
+                                <button type="submit" class="new-comment-submit" v-if="newReply.body"> Post </button>
+                            </form>
+                        </transition>
+
+                        <div class="actions">
+                            <button @click="addReply(post)">
+                                <img src="@/assets/icon-reply.svg">
+                                Reply
+                            </button>
+                            <button v-if="viewType === 'data'" @click="viewDataClick(post)">
+                                <img src="@/assets/icon-view-data.svg">
+                                View Data
+                            </button>
+                        </div>
+                    </div>
+                </div>
+            </transition-group>
+        </div>
+    </section>
+</template>
+
+<script>
```

Comment:
Can you please change this to lang="ts" and add TypeScript return types and parameter types to things? The original code base wasn't TS and I'm slowly trying to migrate everything over.

Reply:
Done